p4-16/spec/P4-16-draft-spec.mdk

```diff
@@ -4038,34 +4039,36 @@ ParserModel {
 }
 ~ End P4Pseudo
 
-The target architecture must specify its behavior for packets that
+An architecture must specify its behavior for packets that
 are ```accept```ed and ```reject```ed. For example, an architecture may
 specify that all packets reaching the ```reject``` state are dropped
-without any further processing. Another architecture may specify that
-such packets are always passed on to the next control block after the
+without further processing. Alternatively, it may specify that
+such packets are passed to the next control block after the
 parser, with intrinsic metadata indicating that the parser reached
-the ```reject``` state, along with any error recorded.
+the ```reject``` state, along with any errors recorded.
 
 ## Parser states { #sec-parser-state-stmt }
 
-A parser state has a name and a body. The body of the state describes
-data processing performed when the parser transitions to the specified
-state. This data processing may consist of:
-
-- Data extraction from packet into headers
-- Invocations of methods of ```extern``` blocks (e.g., checksum
-  computations)
-- Checks for data validity
-- Transition to other states
-
-A state is declared with the following syntax:
+A parser state is declared with the following syntax:
 ~ Begin P4Grammar
 parserState
     : optAnnotations STATE name
      '{' parserStatements transitionStatement '}'
     ;
 ~ End P4Grammar
 
+Each state has a name and a body. The body consists of a sequence of
+statements that describe the processing performed when the parser
+transitions to that state including:
+
+- Mocal variable declarations,
+- Assignment statements, and
+- Method calls, which can serve several purposes:
+  * Invoking functions (e.g., using ```verify``` to check the validity of data already parsed), and
+  * Invoking methods (e.g., extracting data out of packets or computing checksums) and other parsers (see Section [ #sec-invoke-subparser ]).
+
+The syntax for parser statements is given by the following grammar rules:
```

Comment:
You left out transition statements from the list, which should be kept, I think?